File: stock-backend/src/main/java/com/hahn/stock/service/ProductService.java
package com.hahn.stock.service;

import com.hahn.stock.dto.ProductDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProductService {

    ProductDto createProduct(ProductDto productDto);
    ProductDto getProductById(String id);
    List<ProductDto> getAllProducts();
    Page<ProductDto> getAllProducts(Pageable pageable);
    ProductDto updateProduct(String id, ProductDto productDto);
    void deleteProduct(String id);
    List<ProductDto> getProductsByCategory(String categoryId);
}

File: stock-frontend/src/services/productService.js
import api from './api';

export const getProducts = async () => {
    const response = await api.get('/products');
    return response.data;
};

export const getProductsByCategory = async (categoryId) => {
    const response = await api.get(`/products/category/${categoryId}`);
    return response.data;
};

export const getProduct = async (id) => {
    const response = await api.get(`/products/${id}`);
    return response.data;
};

export const createProduct = async (productData) => {
    const response = await api.post('/products', productData);
    return response.data;
};

export const updateProduct = async (id, productData) => {
    const response = await api.put(`/products/${id}`, productData);
    return response.data;
};

export const deleteProduct = async (id) => {
    await api.delete(`/products/${id}`);
};